lib/surface/compiler.ex

```diff
@@ -878,7 +878,8 @@ defmodule Surface.Compiler do
   defp validate_tag_children([]), do: :ok
 
   defp validate_tag_children([%AST.SlotEntry{name: name} | _]) do
-    {:error, "slot entries are only allowed as children elements of components, but found slot entry for #{name}"}
+    {:error,
+     "Slot entries are only allowed as children elements of components. No slot found named slot. Did you mean <##{name} />"}
```

Comment:
I think you need a conditional, something like below. I also cleaned up the grammar of the message

```suggestion
   message =
     case name do
       "slot" -> "No slot found named slot. Did you mean the default slot? Use <#slot />"
       _ -> "Slot entries are only allowed as children elements of components, but found slot entry for #{name}"
     end
    {:error, message}
```

Reply:
Note i haven't tested this, I'm just trying to provide an example

Reply:
but the syntax for named slots is:

```
slot header

~F"""
  <div>
    <:slot name="body" />
  </div>
"""
```

it will match with the "slot" condition in `case`, and I need to track the props attributes values to see if it is a default slot or a named slot. I am thinking in this way:


```elixir
  defp validate_tag_children([%AST.SlotEntry{name: name, props: []} | _]) do
    {:error,
     "Slot entries are only allowed as children elements of components. No slot found named slot. Did you mean <##{name} />"}
  end

  defp validate_tag_children([%AST.SlotEntry{name: name, props: props} | _]) do
    [%Surface.AST.Attribute{value: attrs_values}] = props
    named_value = Map.get(attrs_values, :value, nil)

    {:error,
     "Slot entries are only allowed as children elements of components. No slot found named slot. Did you mean <##{name} name=\"#{named_value}\" />"}
  end
```

WDYT? @davydog187